### PR TITLE
[Text]line break mode for ios

### PR DESCRIPTION
**Pull request by skv-headless**
What do you think is ```lineBreakMode``` a good name? For android it is called ```ellipsize```.

### CODE REVIEW

**Comment by facebook-github-bot-7** [comment]
By analyzing the blame information on this pull request, we identified @nicklockwood and @sahrens to be potential reviewers.

**Comment by facebook-github-bot-9** [comment]
@skv-headless updated the pull request.

**Comment by javache** [comment]
Hey @skv-headless, thanks for this PR, we just needed something like this. What would it take to keep the API consistent with Android, or at least similar?

**Comment by nicklockwood** [comment]
On iOS it looks like we already have an `+[RCTConvert NSLineBreakMode:]` method, but it seems like we aren't using it anywhere.

Please update your PR to use `NSLineBreakMode` instead of `NSString` for the lineBreakMode.

The actual strings we're using in `+[RCTConvert NSLineBreakMode:]` seem rather verbose, and are camelCase instead of kebab-case like our other styles:

    @"wordWrapping": @(NSLineBreakByWordWrapping),
    @"charWrapping": @(NSLineBreakByCharWrapping),
    @"clipping": @(NSLineBreakByClipping),
    @"truncatingHead": @(NSLineBreakByTruncatingHead),
    @"truncatingTail": @(NSLineBreakByTruncatingTail),
    @"truncatingMiddle": @(NSLineBreakByTruncatingMiddle),

We should probably change them. I think "head", "middle" and "tail" are nice, but I'm not sure they adequately cover the available options. How about something like:

    @"word-wrap": @(NSLineBreakByWordWrapping),
    @"char-wrap": @(NSLineBreakByCharWrapping),
    @"clip": @(NSLineBreakByClipping),
    @"truncate-head": @(NSLineBreakByTruncatingHead),
    @"truncate-tail": @(NSLineBreakByTruncatingTail),
    @"truncate-middle": @(NSLineBreakByTruncatingMiddle),

?

**Comment by skv-headless** [comment]
@javache I will try to make it for android. I think ```ellipsize``` working in the same way.

**Comment by nicklockwood** [comment]
I'm actually a little confused about what "word-wrap" does in the context of a single-line label. I guess it probably clips to the nearest whole word?

That's fairly obscure functionality if so, so we might be able to get away with just using:

    @"clip": @(NSLineBreakByClipping),
    @"head": @(NSLineBreakByTruncatingHead),
    @"tail": @(NSLineBreakByTruncatingTail),
    @"middle": @(NSLineBreakByTruncatingMiddle),

And not bother with the other options. What do you think?

**Comment by facebook-github-bot-6** [comment]
@skv-headless updated the pull request.

**Comment by skv-headless** [comment]
@nicklockwood I think better keep only 4 options ```clip```, ```head```, ```tail```,  ```middle```
but I like android naming more ```start```, ```middle```, ```end```.

**Comment by facebook-github-bot-8** [comment]
@skv-headless updated the pull request.

**Comment by skv-headless** [comment]
@nicklockwood I've updated according to your latest note.
@javache it is working for android except ```marquee``` (some problem with focus I suppose). I will try to add this in next PR.

**Comment by facebook-github-bot-8** [comment]
@skv-headless updated the pull request.

**Comment by skv-headless** [comment]
test failures same as in master

**Comment by facebook-github-bot-0** [comment]
@skv-headless updated the pull request.

**Comment by facebook-github-bot-1** [comment]
@skv-headless updated the pull request.

File: Libraries/Text/RCTShadowText.h [hunk 1]
```diff
@@ -24,6 +24,7 @@ extern NSString *const RCTReactTagAttributeName;
 @property (nonatomic, assign) CGFloat letterSpacing;
 @property (nonatomic, assign) CGFloat lineHeight;
 @property (nonatomic, assign) NSUInteger numberOfLines;
+@property (nonatomic, copy) NSString *lineBreakMode;
```
Comment (nicklockwood) [inline Libraries/Text/RCTShadowText.h]: If you define the type as NSLineBreakMode (here and in the RCT_EXPORT_SHADOW_PROPERTY), React will automatically apply the `[RCTConvert NSLineBreakMode:]` function, so there's no need to do it manually.

**Comment by skv-headless** [comment]
@nicklockwood updated

**Comment by nicklockwood** [comment]
@skv-headless awesome! thank you.

**Comment by nicklockwood** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-9** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1113393192036341/int_phab) to review.

**Comment by skv-headless** [comment]
Thank you. With react native I started to love iOS programming 😄

**Pull request closed without merging**